Runar/Runar/Services/Notifications/LocalNotificationManager.swift

```diff
@@ -0,0 +1,117 @@
+//  LocalNotificationManager.swift
+//  Runar
+//
+//  Created by Alexey Poletaev on 09.02.2023.
+//
+
+import UserNotifications
+
+private extension String {
+    static let notificationGeneralContentTitle = L10n.Notifications.General.title
+    static let notificationGeneralContentBody = L10n.Notifications.General.body
+    static let notificationGeneralIdentifier = "General Local Notification"
+}
+
+private enum TimeOfSendingLocalNotifications: Int {
+    case weekday = 2 // Monday
+    case hour = 11   // 11:00
+}
+
+protocol LocalNotificationInterface {
+    /// Checking the resolution of notifications and add notification
+    func addNotification()
+
+    /// Remove all notification
+    func removeAllNotifications()
+}
+
+final class LocalNotificationManager: NSObject, LocalNotificationInterface {
+
+    private let notificationCenter = UNUserNotificationCenter.current()
+
+    /// If the user logs into the app on the day of the notification call before the time of the call, the notification on that day will be deleted
+    private func checkTheDayAndTime() {
+        let dateComponents = Calendar.current.dateComponents([.weekday, .hour, .weekOfMonth], from: Date())
+        let weekdaySetPoint = TimeOfSendingLocalNotifications.weekday.rawValue
+        let hourSetPoint = TimeOfSendingLocalNotifications.hour.rawValue
+
+        guard let weekday = dateComponents.weekday,
+              let hour = dateComponents.hour,
+              let week = dateComponents.weekOfMonth else { return }
+
+        if weekday == weekdaySetPoint && hour < hourSetPoint {
+            var identifier: String = .notificationGeneralIdentifier
+            identifier += "_Week_\(week)"
+            notificationCenter.removePendingNotificationRequests(withIdentifiers: [identifier])
+        }
+    }
+
+    /// Request permission from the user to receive notifications
+    private func requestAuthorization() {
+        notificationCenter.requestAuthorization(options: [.alert, .sound, .badge]) { [weak self] granted, error in
+            guard let self = self else { return }
+
+            if granted {
+                self.addRequest()
+            } else {
+                print("Local Notification: User has declined notifications")
+            }
+        }
+    }
+
+    /// The function of generating sending notifications
+    private func addRequest() {
+        // Notification Content
+        let content = UNMutableNotificationContent()
+        content.title = .notificationGeneralContentTitle
+        content.body = .notificationGeneralContentBody
+        content.sound = UNNotificationSound.default
+        content.badge = 1
+
+        // Configureting the recurring date
+        var dateComponents = DateComponents()
+        dateComponents.calendar = Calendar.current
+        dateComponents.weekday = TimeOfSendingLocalNotifications.weekday.rawValue
+        dateComponents.hour = TimeOfSendingLocalNotifications.hour.rawValue
+
+        for week in 1...5 {
```

Comment:
Looks like, in time we'll see 5 notification one by one

Reply:
One by one, but in different weeks.

Reply:
Honestly, is it your own implementation?

Reply:
Yes this is my implementation